api/v1/error.go

```diff
@@ -0,0 +1,66 @@
+package v1
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// server-side error
+var (
+	ErrGRPCTimeout          = status.New(codes.Unavailable, "trustyserver: request timed out").Err()
+	ErrGRPCPermissionDenied = status.New(codes.PermissionDenied, "trustyserver: permission denied").Err()
+
+	errStringToError = map[string]error{
+		ErrorDesc(ErrGRPCTimeout):          ErrGRPCTimeout,
+		ErrorDesc(ErrGRPCPermissionDenied): ErrGRPCPermissionDenied,
+	}
+)
+
+// client-side error
+var (
+	ErrTimeout          = Error(ErrGRPCTimeout)
+	ErrPermissionDenied = Error(ErrGRPCPermissionDenied)
+)
+
+// TrustyError defines gRPC server errors.
+// (https://github.com/grpc/grpc-go/blob/master/rpc_util.go#L319-L323)
+type TrustyError struct {
+	code codes.Code
+	desc string
+}
+
+// Code returns grpc/codes.Code.
+func (e TrustyError) Code() codes.Code {
+	return e.code
+}
+
+func (e TrustyError) Error() string {
+	return e.desc
+}
+
+// Error returns TrustyError
+func Error(err error) error {
```

Comment:
@dissoupov I kind of disagree with return type being interface. See in general one should thrive having method parameters as generic as possible(interface) but the return type of the method is encouraged to be as specific as possible to that it can later be assigned to any interface reference. In other words. With the current definition, the caller must always do unnecessary check `if _, ok := err.(v1.TrustyError); ok` before using the returned value. But there should not be a need to do that because clearly, the method can only return TrustyError. In other words, I suggest changing the return type to TrustyError. 

```
func Error(err error) TrustyError {
...
}
```

Reply:
Inside the function, if it's not a gRPC error, then return it as is. So it can't be `TrustyError`
```
	if err == nil {
		return nil
	}
	verr, ok := errStringToError[ErrorDesc(err)]
	if !ok {
		// not gRPC error
		return err
	}
```

Reply:
oh i see, did not pay attention to that.